legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/helperFunctions/helperFunctions.pure

```diff
@@ -107,18 +107,43 @@ function meta::relational::functions::database::setUpData(records:List<String>[*
 
    meta::alloy::service::execution::setUpDataSQLs($records, $database);
 }
-
 // This function is needed because split does not preserve empty strings
 // We work around by inserting a non-empty value, doing a split, and then removing the non-empty value
 function <<access.private>> meta::alloy::service::execution::splitWithEmptyValue(s:String[1]) : String[*]
 {
    let delimForEmptyCsvField = '|EMPTY|';
 
    $s->meta::alloy::service::execution::replaceWithEmptyValue($delimForEmptyCsvField)
-        ->split(',')
+        ->meta::alloy::service::execution::splitCsvRow(',')
         ->map(csv_val| if($csv_val == $delimForEmptyCsvField, | '' ,| $csv_val));
 }
 
+Class <<access.private>> meta::alloy::service::execution::SplitterState
+{
+   tokens:String[*];
+   inQuotes:Boolean[1];
+   quoteType: String[1];
+   currentTokenChars:String[*];
+}
+
+function meta::alloy::service::execution::splitCsvRow(s: String[1], delimiter: String[1]): String[*]
+{
+  let charArray = chunk($s, 1);
+  let finalSplitterState = $charArray->fold({currentChar, splitterState|
+      if ($currentChar == '\'' || $currentChar == '\"',
+          |if (!$splitterState.inQuotes,
```

Comment:
`if (!$splitterState.inQuotes && $splitterState.currentTokenChars->makeString()->trim() == '',`

Reply:
Hello @vikask-gs , I reverted this function to keep double quotes like  `(1,\'Peter\',\'"I\'\'m Smith, Jr"\',23,1,1,2);'`
The reason why we need to keep double quotes is that database tables names could have double quotes like what we have in naming.pure 

```
    MyClass: Relational
    {
        a: [myDB] testTable."Column Name with Space",
        b: [myDB] @testJoin,
        d: EnumerationMapping Active : [myDB] testTable."Column&A"
    }
```

If we discard double quotes, we will generate just `Column Name with Space`, which is causing engine build failures.